### PR TITLE
Improve Dockerfile style and Remove pytest-sugar from travis

### PR DESCRIPTION
**Pull request by Drake-Eidukas**
just testing effects of caching pip, as installing python dependencies takes a loooong time

### CODE REVIEW

**Comment by Drake-Eidukas** [comment]
I've cleaned up the branch, now this should just result in a cleaner dockerfile (for maintainability, and also follows best practices), as well as (marginally) faster / better CI (it no longer will be polluted by pytest-sugar).

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,22 @@
 FROM python:3.7.1-stretch
 
-ENV PYTHONUNBUFFERED=1
-
-RUN apt-get update && apt-get install -y vim tmux
+ENV PYTHONUNBUFFERED    1
```
Comment (smkuls) [inline Dockerfile:5]: Since you started working on this, can you add a comment so as to why we need each variable?

File: setup.py [hunk 2]
```diff
@@ -29,7 +29,6 @@
     'pytest-mock',
     'pytest-cov',
     'pytest-env',
-    'pytest-sugar',
     'pytest-xdist',
```
Comment (smkuls) [inline setup.py:32]: Can you get rid of the `-nauto` in `pytest.ini` as well? You should edit the travis files to make the tests run using `-nauto` and it shouldn't be a default option.

Like we discussed lets not run the tests in parallel by default.  I found both sugar and xdist very annoying to work with. With sugar when I tried to use the `-s` switch it wouldn't spit out the logs immediately at times.

File: setup.py [hunk 3]
```diff
@@ -39,6 +38,7 @@
 DEV_REQUIRES = [
     'ipdb',
     'ipython',
+    'pytest-sugar',
```
Comment (smkuls) [inline setup.py]: Ummmm, I don't quite agree with this. This section is for all the packages need for development only. You should make a TEST_OPTIONAL list and then add it there.

**Review by smkuls — DISMISSED** [review]

File: Dockerfile (hunk 1 above)
Comment (smkuls) [reply Dockerfile:5]: so the two paths are needed by `charm`. I think `PYTHONBUFFERED` has got to do something with the buffering of stream when spitting data on to the console.

**Comment by smkuls** [comment]
Can you also create an issue to improve the build times of Travis and add the current time it takes? You can close the issue once this gets merged but it would be good to have it as an issue to look back at what was optimized and how it was optimized.

File: setup.py (hunk 3 above)
Comment (Drake-Eidukas) [reply setup.py]: We should probably also document what each of these sections is for as well for people in the future adding dependencies

Comment (smkuls) [reply setup.py]: I thought the naming and categorization was pretty obvious given we have a separate tests and dev section. But you are right, some comments would be better :)

**Comment by Drake-Eidukas** [comment]
So, it seems like this has gotten travis to start displaying coverage again (resolving #89), but it required me to suppress `DeprecationWarning` errors to get it to show up in the logs (they were drowning them out)

~More work has to be done to get `codecov` to work, namely copying out the coverage report into travis from docker.~

**Comment by Drake-Eidukas** [comment]
That said, this might be a bit out of scope for this PR, so I can split out the stuff to get coverage to work too if you want

File: .ci/travis-script.sh [hunk 4]
```diff
@@ -2,13 +2,21 @@
 
 set -ev
 
-BASE_CMD="docker-compose -f .travis.compose.yml run --rm test-hbmpc"
```
Comment (Drake-Eidukas) [inline .ci/travis-script.sh:5]: Not sure if we need to actually rm the docker container, but if we don't `--rm` it right away, then we can copy over the coverage report for use by `codecov`

**Review by amiller — APPROVED** [review]
I looked through this and reviewed the conversation, looks great to me

**Pull request merged**